maestrowf/datastructures/core/study.py

```diff
@@ -335,7 +365,7 @@ def add_step(self, step):
         :param step: A StudyStep instance to be added to the Study instance.
         """
         # Add the node to the DAG.
-        self.add_node(step.name, step)
+        self.add_node(step.real_name, step)
```

Comment:
why not just use step.name?  won't nickname need to be unique as well for it to work?   Or is it just for easier debugging since hashes aren't the most readable things?

Reply:
That's true -- and I had considered using the hash. It's a mix of wanting to keep things behaving as close to what they were before and the debugging.